app/src/main/java/com/example/campusguide/search/outdoor/PlacesApiSearchResultProvider.kt

```diff
@@ -73,4 +75,32 @@ class PlacesApiSearchResultProvider constructor(activity: Activity, private val
 
         return results
     }
-}
\ No newline at end of file
+
+    suspend fun searchNearbyPlaces(
+        pointOfInterest: String,
+        currentLocation: Location
+    ) = suspendCoroutine<FindAutocompletePredictionsResponse> { cont ->
+
+        val searchBounds = RectangularBounds.newInstance(
+            LatLng(currentLocation.lat - 0.01, currentLocation.lon - 0.01),
+            LatLng(currentLocation.lat + 0.01, currentLocation.lon + 0.01)
+        )
+
+
+        val request = FindAutocompletePredictionsRequest.builder()
+            .setQuery(pointOfInterest)
+            .setTypeFilter(TypeFilter.ESTABLISHMENT)
+            .setLocationRestriction(searchBounds)
+            .build()
+
+        placesClient.findAutocompletePredictions(request).addOnSuccessListener { response ->
+
+            cont.resume(getPlaces(response))
+        }
+    }
+
+    private fun getPlaces(response: FindAutocompletePredictionsResponse): FindAutocompletePredictionsResponse {
+            return(response)
```

Comment:
what is the point of this function? Can't you just cast it inline?

Reply:
I need to be able to return that value but I can't return it from within the Lambda expression so i needed to pop it out